python3Packages.jax[lib]: 0.6.2 -> 0.7.1

GaetanLepage opened the pull request:
## Things done

Diff: https://github.com/jax-ml/jax/compare/jax-v0.6.2...jax-v0.7.1
Changelogs:
- https://github.com/jax-ml/jax/releases/tag/jax-v0.7.0
- https://github.com/jax-ml/jax/releases/tag/jax-v0.7.1
---

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
python3Packages.jax[lib]: 0.6.2 -> 0.7.0

Diff: https://github.com/jax-ml/jax/compare/jax-v0.6.2...jax-v0.7.0

Changelog: https://github.com/jax-ml/jax/releases/tag/jax-v0.7.0

GaetanLepage committed:
python3Packages.tensorflow-probability: patch to fix jax 0.7.0 compatibility

GaetanLepage committed:
python3Packages.chex: remove unused input pythonOlder

samuela reviewed (approved):
diff lgtm

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 427588`
Commit: `e03950bc0713f05a84564cb86068c60fbd46af2d`

---
### `x86_64-linux`
<details>
  <summary>:x: 32 packages failed to build:</summary>
  <ul>
    <li>python312Packages.arviz</li>
    <li>python312Packages.arviz.dist</li>
    <li>python312Packages.bambi</li>
    <li>python312Packages.bambi.dist</li>
    <li>python312Packages.blackjax</li>
    <li>python312Packages.blackjax.dist</li>
    <li>python312Packages.corner</li>
    <li>python312Packages.corner.dist</li>
    <li>python312Packages.distrax</li>
    <li>python312Packages.distrax.dist</li>
    <li>python312Packages.equinox</li>
    <li>python312Packages.equinox.dist</li>
    <li>python312Packages.flowjax</li>
    <li>python312Packages.flowjax.dist</li>
    <li>python312Packages.flowmc</li>
    <li>python312Packages.flowmc.dist</li>
    <li>python312Packages.lineax</li>
    <li>python312Packages.lineax.dist</li>
    <li>python312Packages.numpyro</li>
    <li>python312Packages.numpyro.dist</li>
    <li>python312Packages.nutpie</li>
    <li>python312Packages.nutpie.dist</li>
    <li>python312Packages.optimistix</li>
    <li>python312Packages.optimistix.dist</li>
    <li>python312Packages.paramax</li>
    <li>python312Packages.paramax.dist</li>
    <li>python312Packages.pymc</li>
    <li>python312Packages.pymc.dist</li>
    <li>python312Packages.rlax</li>
    <li>python312Packages.rlax.dist</li>
    <li>python313Packages.blackjax</li>
    <li>python313Packages.blackjax.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 144 packages built:</summary>
  <ul>
    <li>exo</li>
    <li>exo.dist</li>
    <li>python312Packages.ale-py</li>
    <li>python312Packages.ale-py.dist</li>
    <li>python312Packages.anndata</li>
    <li>python312Packages.anndata.dist</li>
    <li>python312Packages.array-api-compat</li>
    <li>python312Packages.array-api-compat.dist</li>
    <li>python312Packages.augmax</li>
    <li>python312Packages.augmax.dist</li>
    <li>python312Packages.brax</li>
    <li>python312Packages.brax.dist</li>
    <li>python312Packages.chex</li>
    <li>python312Packages.chex.dist</li>
    <li>python312Packages.cleanlab</li>
    <li>python312Packages.cleanlab.dist</li>
    <li>python312Packages.dalle-mini</li>
    <li>python312Packages.dalle-mini.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.dm-haiku</li>
    <li>python312Packages.dm-haiku.dist</li>
    <li>python312Packages.dm-sonnet</li>
    <li>python312Packages.dm-sonnet.dist</li>
    <li>python312Packages.edward</li>
    <li>python312Packages.edward.dist</li>
    <li>python312Packages.evosax</li>
    <li>python312Packages.evosax.dist</li>
    <li>python312Packages.flax</li>
    <li>python312Packages.flax.dist</li>
    <li>python312Packages.gymnasium</li>
    <li>python312Packages.gymnasium.dist</li>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax-cuda12-pjrt</li>
    <li>python312Packages.jax-cuda12-plugin</li>
    <li>python312Packages.jax.dist</li>
    <li>python312Packages.jaxlib (python312Packages.jaxlib-bin)</li>
    <li>python312Packages.jaxlib.dist (python312Packages.jaxlib-bin.dist)</li>
    <li>python312Packages.jaxopt</li>
    <li>python312Packages.jaxopt.dist</li>
    <li>python312Packages.jmp</li>
    <li>python312Packages.jmp.dist</li>
    <li>python312Packages.keras</li>
    <li>python312Packages.keras.dist</li>
    <li>python312Packages.mhcflurry</li>
    <li>python312Packages.mhcflurry.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.mujoco-mjx</li>
    <li>python312Packages.mujoco-mjx.dist</li>
    <li>python312Packages.nifty8</li>
    <li>python312Packages.nifty8.dist</li>
    <li>python312Packages.optax</li>
    <li>python312Packages.optax.dist</li>
    <li>python312Packages.optax.testsout</li>
    <li>python312Packages.orbax-checkpoint</li>
    <li>python312Packages.orbax-checkpoint.dist</li>
    <li>python312Packages.pettingzoo</li>
    <li>python312Packages.pettingzoo.dist</li>
    <li>python312Packages.pymanopt</li>
    <li>python312Packages.pymanopt.dist</li>
    <li>python312Packages.pytensor</li>
    <li>python312Packages.pytensor.dist</li>
    <li>python312Packages.pyvers</li>
    <li>python312Packages.pyvers.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.skrl</li>
    <li>python312Packages.skrl.dist</li>
    <li>python312Packages.stable-baselines3</li>
    <li>python312Packages.stable-baselines3.dist</li>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python312Packages.tensorflow-datasets</li>
    <li>python312Packages.tensorflow-datasets.dist</li>
    <li>python312Packages.tensorflow-probability</li>
    <li>python312Packages.tensorflow-probability.dist</li>
    <li>python312Packages.tianshou</li>
    <li>python312Packages.tianshou.dist</li>
    <li>python312Packages.tinygrad</li>
    <li>python312Packages.tinygrad.dist</li>
    <li>python312Packages.treescope</li>
    <li>python312Packages.treescope.dist</li>
    <li>python312Packages.tyro</li>
    <li>python312Packages.tyro.dist</li>
    <li>python312Packages.unsloth</li>
    <li>python312Packages.unsloth-zoo</li>
    <li>python312Packages.unsloth-zoo.dist</li>
    <li>python312Packages.unsloth.dist</li>
    <li>python312Packages.viser</li>
    <li>python312Packages.viser.dist</li>
    <li>python312Packages.vqgan-jax</li>
    <li>python312Packages.vqgan-jax.dist</li>
    <li>python312Packages.waymax</li>
    <li>python312Packages.waymax.dist</li>
    <li>python313Packages.anndata</li>
    <li>python313Packages.anndata.dist</li>
    <li>python313Packages.array-api-compat</li>
    <li>python313Packages.array-api-compat.dist</li>
    <li>python313Packages.augmax</li>
    <li>python313Packages.augmax.dist</li>
    <li>python313Packages.chex</li>
    <li>python313Packages.chex.dist</li>
    <li>python313Packages.equinox</li>
    <li>python313Packages.equinox.dist</li>
    <li>python313Packages.flowmc</li>
    <li>python313Packages.flowmc.dist</li>
    <li>python313Packages.jax</li>
    <li>python313Packages.jax-cuda12-pjrt</li>
    <li>python313Packages.jax-cuda12-plugin</li>
    <li>python313Packages.jax.dist</li>
    <li>python313Packages.jaxlib (python313Packages.jaxlib-bin)</li>
    <li>python313Packages.jaxlib.dist (python313Packages.jaxlib-bin.dist)</li>
    <li>python313Packages.jaxopt</li>
    <li>python313Packages.jaxopt.dist</li>
    <li>python313Packages.jmp</li>
    <li>python313Packages.jmp.dist</li>
    <li>python313Packages.lineax</li>
    <li>python313Packages.lineax.dist</li>
    <li>python313Packages.mujoco-mjx</li>
    <li>python313Packages.mujoco-mjx.dist</li>
    <li>python313Packages.nifty8</li>
    <li>python313Packages.nifty8.dist</li>
    <li>python313Packages.optax</li>
    <li>python313Packages.optax.dist</li>
    <li>python313Packages.optax.testsout</li>
    <li>python313Packages.optimistix</li>
    <li>python313Packages.optimistix.dist</li>
    <li>python313Packages.orbax-checkpoint</li>
    <li>python313Packages.orbax-checkpoint.dist</li>
    <li>python313Packages.paramax</li>
    <li>python313Packages.paramax.dist</li>
    <li>python313Packages.pyvers</li>
    <li>python313Packages.pyvers.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
    <li>python313Packages.tinygrad</li>
    <li>python313Packages.tinygrad.dist</li>
    <li>python313Packages.treescope</li>
    <li>python313Packages.treescope.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.blackjax</summary>
<pre>  772  773  774  775  776  777  778  779  780  781  782  783  784  785
  786  787  788  789  790  791  792  793  794  795  796  797  798  799
  800  801  802  803  804  805  806  807  808  809  810  811  812  813
  814  815  816  817  818  819  820  821  822  823  824  825  826  827
  828  829  830  831  832  833  834  835  836  837  838  839  840  841
  842  843  844  845  846  847  848  849  850  851  852  853  854  855
  856  857  858  859  860  861  862  863  864  865  866  867  868  869
  870  871  872  873  874  875  876  877  878  879  880  881  882  883
  884  885  886  887  888  889  890  891  892  893  894  895  896  897
  898  899  900  901  902  903  904  905  906  907  908  909  910  911
  912  913  914  915  916  917  918  919  920  921  922  923  924  925
  926  927  928  929  930  931  932  933  934  935  936  937  938  939
  940  941  942  943  944  945  946  947  948  949  950  951  952  953
  954  955  956  957  958  959  960  961  962  963  964  965  966  967
  968  969  970  971  972  973  974  975  976  977  978  979  980  981
  982  983  984  985  986  987  988  989  990  991  992  993  994  995
  996  997  998  999 1000 1001]
=========================== short test summary info ============================
FAILED tests/adaptation/test_adaptation.py::test_chees_adaptation[adaptation_filters0] - AssertionError: 
============= 1 failed, 461 passed, 1 skipped in 68.17s (0:01:08) ==============</pre>
</details>
<details>
<summary>python312Packages.distrax</summary>
<pre>FAILED distrax/_src/distributions/quantized_test.py::QuantizedSurvivalFunctionConsistencyTest::test_log_survival_function_log_cdf_consistency_noop cutoffs__with_device - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/distributions/von_mises_test.py::VonMisesTest::test_with_two_distributions_cross-ent distrax_to_distrax__without_device - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/importance_sampling_test.py::ImportanceSamplingTest::test_importance_sampling_ratios_off_policy__without_device - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/importance_sampling_test.py::ImportanceSamplingTest::test_importance_sampling_ratios_off_policy__with_jit - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/distributions/von_mises_test.py::VonMisesTest::test_with_two_distributions_kl distrax_to_distrax__with_jit - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/jittable_test.py::JittableTest::test_traceable - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/distributions/von_mises_test.py::VonMisesTest::test_with_two_distributions_cross-ent distrax_to_distrax__with_device - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/importance_sampling_test.py::ImportanceSamplingTest::test_importance_sampling_ratios_on_policy__with_device - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/jittable_test.py::JittableTest::test_different_jittables_to_compiled_function - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/importance_sampling_test.py::ImportanceSamplingTest::test_importance_sampling_ratios_on_policy__without_device - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/jittable_test.py::JittableTest::test_jittable - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/jittable_test.py::JittableTest::test_modifying_object_data_does_not_leak_tracers - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/importance_sampling_test.py::ImportanceSamplingTest::test_importance_sampling_ratios_on_policy__with_jit - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/distributions/von_mises_test.py::VonMisesTest::test_with_two_distributions_cross-ent distrax_to_distrax__with_jit - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/jittable_test.py::JittableTest::test_metadata_modification_statements_are_removed_by_compilation - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/distributions/von_mises_test.py::VonMisesTest::test_with_two_distributions_kl distrax_to_distrax__with_device - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/distributions/von_mises_test.py::VonMisesTest::test_with_two_distributions_kl distrax_to_distrax__without_device - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/jittable_test.py::JittableTest::test_vmappable - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
FAILED distrax/_src/utils/importance_sampling_test.py::ImportanceSamplingTest::test_importance_sampling_ratios_off_policy__with_device - AttributeError: jax.interpreters.xla.abstractify was deprecated in JAX v0.5...
== 402 failed, 5313 passed, 1025 skipped, 2323 warnings in 183.23s (0:03:03) ===</pre>
</details>
<details>
<summary>python312Packages.equinox</summary>
<pre>            # Linearize so that we save residuals
            return jax.linearize(loop, init_xs)
    
        size = 100_000
        # nontrivial batch size is important to ensure that the `.at[].set()` is really a
        # scatter, and that XLA doesn't optimise it into a dynamic_update_slice. (Which
        # can be switched with `select` in the compiler.)
        args = jnp.array([0, 1]), jnp.zeros((2, size))
        f(*args)  # compile
    
        speed = timeit.timeit(lambda: f(*args), number=1)
>       assert speed < 4
E       assert 4.725688154809177 < 4

tests/test_while_loop.py:514: AssertionError
----------------------------- Captured stdout call -----------------------------
stop_gradient_perturb_val (False, False, False, (False, True))
=========================== short test summary info ============================
FAILED tests/test_while_loop.py::test_speed_buffer_while[True] - assert 4.725688154809177 < 4
============= 1 failed, 452 passed, 1 skipped in 70.17s (0:01:10) ==============</pre>
</details>
<details>
<summary>python313Packages.blackjax</summary>
<pre>  772  773  774  775  776  777  778  779  780  781  782  783  784  785
  786  787  788  789  790  791  792  793  794  795  796  797  798  799
  800  801  802  803  804  805  806  807  808  809  810  811  812  813
  814  815  816  817  818  819  820  821  822  823  824  825  826  827
  828  829  830  831  832  833  834  835  836  837  838  839  840  841
  842  843  844  845  846  847  848  849  850  851  852  853  854  855
  856  857  858  859  860  861  862  863  864  865  866  867  868  869
  870  871  872  873  874  875  876  877  878  879  880  881  882  883
  884  885  886  887  888  889  890  891  892  893  894  895  896  897
  898  899  900  901  902  903  904  905  906  907  908  909  910  911
  912  913  914  915  916  917  918  919  920  921  922  923  924  925
  926  927  928  929  930  931  932  933  934  935  936  937  938  939
  940  941  942  943  944  945  946  947  948  949  950  951  952  953
  954  955  956  957  958  959  960  961  962  963  964  965  966  967
  968  969  970  971  972  973  974  975  976  977  978  979  980  981
  982  983  984  985  986  987  988  989  990  991  992  993  994  995
  996  997  998  999 1000 1001]
=========================== short test summary info ============================
FAILED tests/adaptation/test_adaptation.py::test_chees_adaptation[adaptation_filters0] - AssertionError: 
============= 1 failed, 461 passed, 1 skipped in 188.52s (0:03:08) =============</pre>
</details>
</details>

GaetanLepage committed:
python3Packages.blackjax: skip failing test

https://github.com/blackjax-devs/blackjax/issues/795

GaetanLepage committed:
python3Packages.distrax: fetch upstream patches to fix jax 0.7.0 compatibility

GaetanLepage committed:
python3Packages.jax[lib]: 0.7.0 -> 0.7.1

Diff: https://github.com/jax-ml/jax/compare/jax-v0.7.0...jax-v0.7.1

Changelog: https://github.com/jax-ml/jax/releases/tag/jax-v0.7.1

samuela commented:
Yeah, I also find that peculiar. Not your fault at least

GaetanLepage commented:
> Yeah, I also find that peculiar. Not your fault at least

Reported upstream: https://github.com/jax-ml/jax/issues/31428

And I skipped the relevant test files on darwin.

GaetanLepage commented on pkgs/development/python-modules/jax-cuda12-pjrt/default.nix:
I'm not sure it's ready yet. The current version is `0.0.1rc0`: https://pypi.org/project/jax-cuda13-pjrt/

NickCao reviewed (commented):
Diff LGTM, please run nixpkgs-review.

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 427588 --package python312Packages.jax --package python3Packages.jax`
Commit: `ce3dde9506e91b675dcc1bc05c1b0819436cb6af`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax.dist (python312Packages.jax.dist.dist)</li>
    <li>python3Packages.jax</li>
    <li>python3Packages.jax.dist (python3Packages.jax.dist.dist)</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax.dist (python312Packages.jax.dist.dist)</li>
    <li>python3Packages.jax</li>
    <li>python3Packages.jax.dist (python3Packages.jax.dist.dist)</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax.dist</li>
    <li>python3Packages.jax</li>
    <li>python3Packages.jax.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax.dist (python312Packages.jax.dist.dist)</li>
    <li>python3Packages.jax</li>
    <li>python3Packages.jax.dist (python3Packages.jax.dist.dist)</li>
  </ul>
</details>

GaetanLepage committed:
python3Packages.equinox: backport patches to fix with jax 0.7.0

https://github.com/patrick-kidger/equinox/pull/1086
https://github.com/patrick-kidger/equinox/pull/1082

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 427588`
Commit: `fd62135cc750ab3238c9bdae86a56eb4f327c306`

---
### `x86_64-linux`
<details>
  <summary>:x: 56 packages failed to build:</summary>
  <ul>
    <li>python312Packages.ale-py</li>
    <li>python312Packages.ale-py.dist</li>
    <li>python312Packages.brax</li>
    <li>python312Packages.brax.dist</li>
    <li>python312Packages.cleanlab</li>
    <li>python312Packages.cleanlab.dist</li>
    <li>python312Packages.dalle-mini</li>
    <li>python312Packages.dalle-mini.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.dm-sonnet</li>
    <li>python312Packages.dm-sonnet.dist</li>
    <li>python312Packages.edward</li>
    <li>python312Packages.edward.dist</li>
    <li>python312Packages.evosax</li>
    <li>python312Packages.evosax.dist</li>
    <li>python312Packages.flax</li>
    <li>python312Packages.flax.dist</li>
    <li>python312Packages.gymnasium</li>
    <li>python312Packages.gymnasium.dist</li>
    <li>python312Packages.keras</li>
    <li>python312Packages.keras.dist</li>
    <li>python312Packages.mhcflurry</li>
    <li>python312Packages.mhcflurry.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.optimistix</li>
    <li>python312Packages.optimistix.dist</li>
    <li>python312Packages.pettingzoo</li>
    <li>python312Packages.pettingzoo.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.skrl</li>
    <li>python312Packages.skrl.dist</li>
    <li>python312Packages.stable-baselines3</li>
    <li>python312Packages.stable-baselines3.dist</li>
    <li>python312Packages.tianshou</li>
    <li>python312Packages.tianshou.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python312Packages.tyro</li>
    <li>python312Packages.tyro.dist</li>
    <li>python312Packages.unsloth</li>
    <li>python312Packages.unsloth-zoo</li>
    <li>python312Packages.unsloth-zoo.dist</li>
    <li>python312Packages.unsloth.dist</li>
    <li>python312Packages.vqgan-jax</li>
    <li>python312Packages.vqgan-jax.dist</li>
    <li>python312Packages.waymax</li>
    <li>python312Packages.waymax.dist</li>
    <li>python313Packages.optimistix</li>
    <li>python313Packages.optimistix.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 108 packages built:</summary>
  <ul>
    <li>exo</li>
    <li>exo.dist</li>
    <li>python312Packages.anndata</li>
    <li>python312Packages.anndata.dist</li>
    <li>python312Packages.array-api-compat</li>
    <li>python312Packages.array-api-compat.dist</li>
    <li>python312Packages.augmax</li>
    <li>python312Packages.augmax.dist</li>
    <li>python312Packages.blackjax</li>
    <li>python312Packages.blackjax.dist</li>
    <li>python312Packages.chex</li>
    <li>python312Packages.chex.dist</li>
    <li>python312Packages.dm-haiku</li>
    <li>python312Packages.dm-haiku.dist</li>
    <li>python312Packages.equinox</li>
    <li>python312Packages.equinox.dist</li>
    <li>python312Packages.flowmc</li>
    <li>python312Packages.flowmc.dist</li>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax-cuda12-pjrt</li>
    <li>python312Packages.jax-cuda12-plugin</li>
    <li>python312Packages.jax.dist</li>
    <li>python312Packages.jaxlib (python312Packages.jaxlib-bin)</li>
    <li>python312Packages.jaxlib.dist (python312Packages.jaxlib-bin.dist)</li>
    <li>python312Packages.jaxopt</li>
    <li>python312Packages.jaxopt.dist</li>
    <li>python312Packages.jmp</li>
    <li>python312Packages.jmp.dist</li>
    <li>python312Packages.kserve</li>
    <li>python312Packages.kserve.dist</li>
    <li>python312Packages.lineax</li>
    <li>python312Packages.lineax.dist</li>
    <li>python312Packages.mujoco-mjx</li>
    <li>python312Packages.mujoco-mjx.dist</li>
    <li>python312Packages.nifty8</li>
    <li>python312Packages.nifty8.dist</li>
    <li>python312Packages.optax</li>
    <li>python312Packages.optax.dist</li>
    <li>python312Packages.optax.testsout</li>
    <li>python312Packages.orbax-checkpoint</li>
    <li>python312Packages.orbax-checkpoint.dist</li>
    <li>python312Packages.outlines</li>
    <li>python312Packages.outlines.dist</li>
    <li>python312Packages.paramax</li>
    <li>python312Packages.paramax.dist</li>
    <li>python312Packages.pymanopt</li>
    <li>python312Packages.pymanopt.dist</li>
    <li>python312Packages.pyvers</li>
    <li>python312Packages.pyvers.dist</li>
    <li>python312Packages.scanpy</li>
    <li>python312Packages.scanpy.dist</li>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python312Packages.tensorflow-datasets</li>
    <li>python312Packages.tensorflow-datasets.dist</li>
    <li>python312Packages.tinygrad</li>
    <li>python312Packages.tinygrad.dist</li>
    <li>python312Packages.treescope</li>
    <li>python312Packages.treescope.dist</li>
    <li>vllm (python312Packages.vllm)</li>
    <li>vllm.dist (python312Packages.vllm.dist)</li>
    <li>python313Packages.anndata</li>
    <li>python313Packages.anndata.dist</li>
    <li>python313Packages.array-api-compat</li>
    <li>python313Packages.array-api-compat.dist</li>
    <li>python313Packages.augmax</li>
    <li>python313Packages.augmax.dist</li>
    <li>python313Packages.blackjax</li>
    <li>python313Packages.blackjax.dist</li>
    <li>python313Packages.chex</li>
    <li>python313Packages.chex.dist</li>
    <li>python313Packages.equinox</li>
    <li>python313Packages.equinox.dist</li>
    <li>python313Packages.flowmc</li>
    <li>python313Packages.flowmc.dist</li>
    <li>python313Packages.jax</li>
    <li>python313Packages.jax-cuda12-pjrt</li>
    <li>python313Packages.jax-cuda12-plugin</li>
    <li>python313Packages.jax.dist</li>
    <li>python313Packages.jaxlib (python313Packages.jaxlib-bin)</li>
    <li>python313Packages.jaxlib.dist (python313Packages.jaxlib-bin.dist)</li>
    <li>python313Packages.jaxopt</li>
    <li>python313Packages.jaxopt.dist</li>
    <li>python313Packages.jmp</li>
    <li>python313Packages.jmp.dist</li>
    <li>python313Packages.lineax</li>
    <li>python313Packages.lineax.dist</li>
    <li>python313Packages.mujoco-mjx</li>
    <li>python313Packages.mujoco-mjx.dist</li>
    <li>python313Packages.nifty8</li>
    <li>python313Packages.nifty8.dist</li>
    <li>python313Packages.optax</li>
    <li>python313Packages.optax.dist</li>
    <li>python313Packages.optax.testsout</li>
    <li>python313Packages.orbax-checkpoint</li>
    <li>python313Packages.orbax-checkpoint.dist</li>
    <li>python313Packages.paramax</li>
    <li>python313Packages.paramax.dist</li>
    <li>python313Packages.pyvers</li>
    <li>python313Packages.pyvers.dist</li>
    <li>python313Packages.scanpy</li>
    <li>python313Packages.scanpy.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
    <li>python313Packages.tinygrad</li>
    <li>python313Packages.tinygrad.dist</li>
    <li>python313Packages.treescope</li>
    <li>python313Packages.treescope.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.keras</summary>
<pre>+ float64
FAILED keras/src/ops/numpy_test.py::NumpyDtypeTest::test_kaiser_int64 - AssertionError: 
- float32
+ float64
FAILED keras/src/ops/numpy_test.py::NumpyDtypeTest::test_kaiser_int8 - AssertionError: 
- float32
+ float64
FAILED keras/src/ops/numpy_test.py::NumpyDtypeTest::test_kaiser_none - AssertionError: 
- float32
+ float64
FAILED keras/src/ops/numpy_test.py::NumpyDtypeTest::test_kaiser_uint16 - AssertionError: 
- float32
+ float64
FAILED keras/src/ops/numpy_test.py::NumpyDtypeTest::test_kaiser_uint32 - AssertionError: 
- float32
+ float64
FAILED keras/src/ops/numpy_test.py::NumpyDtypeTest::test_kaiser_uint8 - AssertionError: 
- float32
+ float64
= 74 failed, 13284 passed, 310 skipped, 5 deselected, 4 xfailed, 6 xpassed in 844.10s (0:14:04) =</pre>
</details>
<details>
<summary>python312Packages.optimistix</summary>
<pre>        with context:
            optx_argmin = optx.least_squares(
                fn, solver, init, has_aux=has_aux, args=args, max_steps=10_000, throw=False
            ).value
        out = fn(optx_argmin, args)
        if has_aux:
            residual, _ = out
        else:
            residual = out
        optx_min = jtu.tree_reduce(
            lambda x, y: x + y, jtu.tree_map(lambda x: jnp.sum(x**2), residual)
        )
>       assert tree_allclose(optx_min, minimum, atol=atol, rtol=rtol)
E       assert Array(False, dtype=bool)
E        +  where Array(False, dtype=bool) = tree_allclose(Array(0.12993518, dtype=float64), Array(0., dtype=float64, weak_type=True), atol=0.0001, rtol=0.0001)

tests/test_least_squares.py:53: AssertionError
=========================== short test summary info ============================
FAILED tests/test_least_squares.py::test_least_squares[rosenbrock-minimum2-init2-args2-solver15] - assert Array(False, dtype=bool)
============= 1 failed, 399 passed, 1 skipped in 71.82s (0:01:11) ==============</pre>
</details>
<details>
<summary>python313Packages.optimistix</summary>
<pre>        with context:
            optx_argmin = optx.least_squares(
                fn, solver, init, has_aux=has_aux, args=args, max_steps=10_000, throw=False
            ).value
        out = fn(optx_argmin, args)
        if has_aux:
            residual, _ = out
        else:
            residual = out
        optx_min = jtu.tree_reduce(
            lambda x, y: x + y, jtu.tree_map(lambda x: jnp.sum(x**2), residual)
        )
>       assert tree_allclose(optx_min, minimum, atol=atol, rtol=rtol)
E       assert Array(False, dtype=bool)
E        +  where Array(False, dtype=bool) = tree_allclose(Array(0.12993518, dtype=float64), Array(0., dtype=float64, weak_type=True), atol=0.0001, rtol=0.0001)

tests/test_least_squares.py:53: AssertionError
=========================== short test summary info ============================
FAILED tests/test_least_squares.py::test_least_squares[rosenbrock-minimum2-init2-args2-solver15] - assert Array(False, dtype=bool)
============= 1 failed, 399 passed, 1 skipped in 74.83s (0:01:14) ==============</pre>
</details>
</details>

GaetanLepage committed:
python3Packages.keras: skip failing tests

https://github.com/keras-team/keras/issues/21629

GaetanLepage committed:
python3Packages.optimistix: disable failing test

https://github.com/patrick-kidger/optimistix/issues/160

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 427588`
Commit: `37aa265ca7a7e2e55acc946b755858cdc08900c1`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.cleanlab</li>
    <li>python312Packages.cleanlab.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 162 packages built:</summary>
  <ul>
    <li>exo</li>
    <li>exo.dist</li>
    <li>python312Packages.ale-py</li>
    <li>python312Packages.ale-py.dist</li>
    <li>python312Packages.anndata</li>
    <li>python312Packages.anndata.dist</li>
    <li>python312Packages.array-api-compat</li>
    <li>python312Packages.array-api-compat.dist</li>
    <li>python312Packages.augmax</li>
    <li>python312Packages.augmax.dist</li>
    <li>python312Packages.blackjax</li>
    <li>python312Packages.blackjax.dist</li>
    <li>python312Packages.brax</li>
    <li>python312Packages.brax.dist</li>
    <li>python312Packages.chex</li>
    <li>python312Packages.chex.dist</li>
    <li>python312Packages.dalle-mini</li>
    <li>python312Packages.dalle-mini.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.dm-haiku</li>
    <li>python312Packages.dm-haiku.dist</li>
    <li>python312Packages.dm-sonnet</li>
    <li>python312Packages.dm-sonnet.dist</li>
    <li>python312Packages.edward</li>
    <li>python312Packages.edward.dist</li>
    <li>python312Packages.equinox</li>
    <li>python312Packages.equinox.dist</li>
    <li>python312Packages.evosax</li>
    <li>python312Packages.evosax.dist</li>
    <li>python312Packages.flax</li>
    <li>python312Packages.flax.dist</li>
    <li>python312Packages.flowmc</li>
    <li>python312Packages.flowmc.dist</li>
    <li>python312Packages.gymnasium</li>
    <li>python312Packages.gymnasium.dist</li>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax-cuda12-pjrt</li>
    <li>python312Packages.jax-cuda12-plugin</li>
    <li>python312Packages.jax.dist</li>
    <li>python312Packages.jaxlib (python312Packages.jaxlib-bin)</li>
    <li>python312Packages.jaxlib.dist (python312Packages.jaxlib-bin.dist)</li>
    <li>python312Packages.jaxopt</li>
    <li>python312Packages.jaxopt.dist</li>
    <li>python312Packages.jmp</li>
    <li>python312Packages.jmp.dist</li>
    <li>python312Packages.keras</li>
    <li>python312Packages.keras.dist</li>
    <li>python312Packages.kserve</li>
    <li>python312Packages.kserve.dist</li>
    <li>python312Packages.lineax</li>
    <li>python312Packages.lineax.dist</li>
    <li>python312Packages.mhcflurry</li>
    <li>python312Packages.mhcflurry.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.mujoco-mjx</li>
    <li>python312Packages.mujoco-mjx.dist</li>
    <li>python312Packages.nifty8</li>
    <li>python312Packages.nifty8.dist</li>
    <li>python312Packages.optax</li>
    <li>python312Packages.optax.dist</li>
    <li>python312Packages.optax.testsout</li>
    <li>python312Packages.optimistix</li>
    <li>python312Packages.optimistix.dist</li>
    <li>python312Packages.orbax-checkpoint</li>
    <li>python312Packages.orbax-checkpoint.dist</li>
    <li>python312Packages.outlines</li>
    <li>python312Packages.outlines.dist</li>
    <li>python312Packages.paramax</li>
    <li>python312Packages.paramax.dist</li>
    <li>python312Packages.pettingzoo</li>
    <li>python312Packages.pettingzoo.dist</li>
    <li>python312Packages.pymanopt</li>
    <li>python312Packages.pymanopt.dist</li>
    <li>python312Packages.pyvers</li>
    <li>python312Packages.pyvers.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
    <li>python312Packages.scanpy</li>
    <li>python312Packages.scanpy.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.skrl</li>
    <li>python312Packages.skrl.dist</li>
    <li>python312Packages.stable-baselines3</li>
    <li>python312Packages.stable-baselines3.dist</li>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python312Packages.tensorflow-datasets</li>
    <li>python312Packages.tensorflow-datasets.dist</li>
    <li>python312Packages.tianshou</li>
    <li>python312Packages.tianshou.dist</li>
    <li>python312Packages.tinygrad</li>
    <li>python312Packages.tinygrad.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python312Packages.treescope</li>
    <li>python312Packages.treescope.dist</li>
    <li>python312Packages.tyro</li>
    <li>python312Packages.tyro.dist</li>
    <li>python312Packages.unsloth</li>
    <li>python312Packages.unsloth-zoo</li>
    <li>python312Packages.unsloth-zoo.dist</li>
    <li>python312Packages.unsloth.dist</li>
    <li>vllm (python312Packages.vllm)</li>
    <li>vllm.dist (python312Packages.vllm.dist)</li>
    <li>python312Packages.vqgan-jax</li>
    <li>python312Packages.vqgan-jax.dist</li>
    <li>python312Packages.waymax</li>
    <li>python312Packages.waymax.dist</li>
    <li>python313Packages.anndata</li>
    <li>python313Packages.anndata.dist</li>
    <li>python313Packages.array-api-compat</li>
    <li>python313Packages.array-api-compat.dist</li>
    <li>python313Packages.augmax</li>
    <li>python313Packages.augmax.dist</li>
    <li>python313Packages.blackjax</li>
    <li>python313Packages.blackjax.dist</li>
    <li>python313Packages.chex</li>
    <li>python313Packages.chex.dist</li>
    <li>python313Packages.equinox</li>
    <li>python313Packages.equinox.dist</li>
    <li>python313Packages.flowmc</li>
    <li>python313Packages.flowmc.dist</li>
    <li>python313Packages.jax</li>
    <li>python313Packages.jax-cuda12-pjrt</li>
    <li>python313Packages.jax-cuda12-plugin</li>
    <li>python313Packages.jax.dist</li>
    <li>python313Packages.jaxlib (python313Packages.jaxlib-bin)</li>
    <li>python313Packages.jaxlib.dist (python313Packages.jaxlib-bin.dist)</li>
    <li>python313Packages.jaxopt</li>
    <li>python313Packages.jaxopt.dist</li>
    <li>python313Packages.jmp</li>
    <li>python313Packages.jmp.dist</li>
    <li>python313Packages.lineax</li>
    <li>python313Packages.lineax.dist</li>
    <li>python313Packages.mujoco-mjx</li>
    <li>python313Packages.mujoco-mjx.dist</li>
    <li>python313Packages.nifty8</li>
    <li>python313Packages.nifty8.dist</li>
    <li>python313Packages.optax</li>
    <li>python313Packages.optax.dist</li>
    <li>python313Packages.optax.testsout</li>
    <li>python313Packages.optimistix</li>
    <li>python313Packages.optimistix.dist</li>
    <li>python313Packages.orbax-checkpoint</li>
    <li>python313Packages.orbax-checkpoint.dist</li>
    <li>python313Packages.paramax</li>
    <li>python313Packages.paramax.dist</li>
    <li>python313Packages.pyvers</li>
    <li>python313Packages.pyvers.dist</li>
    <li>python313Packages.scanpy</li>
    <li>python313Packages.scanpy.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
    <li>python313Packages.tinygrad</li>
    <li>python313Packages.tinygrad.dist</li>
    <li>python313Packages.treescope</li>
    <li>python313Packages.treescope.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.cleanlab</summary>
<pre>ERROR tests/datalab/test_issue_finder.py::TestIssueFinder::test_underperforming_group_availability_issue_1065[keys=['pred_probs', 'features', 'cluster_ids'] -> available] - AssertionError
ERROR tests/datalab/test_issue_finder.py::TestIssueFinder::test_underperforming_group_availability_issue_1065[keys=['pred_probs', 'knn_graph', 'cluster_ids'] -> available] - AssertionError
ERROR tests/datalab/test_issue_finder.py::TestIssueFinder::test_underperforming_group_availability_issue_1065[keys=['pred_probs', 'features', 'knn_graph', 'cluster_ids'] -> available] - AssertionError
ERROR tests/datalab/test_issue_finder.py::TestIssueFinder::test_underperforming_group_availability_issue_1065[keys=['pred_probs'] -> unavailable] - AssertionError
ERROR tests/datalab/test_issue_finder.py::TestIssueFinder::test_get_available_issue_types - AssertionError
ERROR tests/datalab/test_issue_finder.py::TestIssueFinder::test_find_issues - AssertionError
ERROR tests/datalab/test_issue_finder.py::TestIssueFinder::test_validate_issue_types_dict - AssertionError
ERROR tests/datalab/test_issue_finder.py::TestRegressionIssueFinder::test_get_available_issue_types - AssertionError
ERROR tests/datalab/test_report.py::TestReporter::test_init - AssertionError
ERROR tests/datalab/test_report.py::TestReporter::test_report - AssertionError
ERROR tests/datalab/test_report.py::TestReporter::test_get_report[True] - AssertionError
ERROR tests/datalab/test_report.py::TestReporter::test_get_report[False] - AssertionError
ERROR tests/datalab/test_report.py::TestReporter::test_show_all_issues[True-Here is a lab summary\n\nfoo report\n\n\nbar report] - AssertionError
ERROR tests/datalab/test_report.py::TestReporter::test_show_all_issues[False-Here is a lab summary\n\nfoo report] - AssertionError
ERROR tests/datalab/test_report.py::TestReporter::test_summary_with_score - AssertionError
ERROR tests/datalab/test_report.py::TestReporter::test_summary_without_score - AssertionError
ERROR tests/spurious_correlation/test_spurious_correlation.py::TestImagelabReporterAdapter::test_report[dark] - AssertionError
ERROR tests/spurious_correlation/test_spurious_correlation.py::TestImagelabReporterAdapter::test_report[odd_aspect_ratio] - AssertionError
ERROR tests/spurious_correlation/test_spurious_correlation.py::TestImagelabReporterAdapter::test_report[identity] - AssertionError
= 151 failed, 652 passed, 15 deselected, 2 xfailed, 26968 warnings, 233 errors in 578.79s (0:09:38) =</pre>
</details>
</details>

samuela commented on pkgs/development/python-modules/distrax/default.nix:
```diff
@@ -31,8 +31,23 @@ buildPythonPackage rec {
       hash = "sha256-hFNXKoA1b5I6dzhwTRXp/SnkHv89GI6tYwlnBBHwG78=";
     })
+    # https://github.com/google-deepmind/distrax/pull/289
+    (fetchpatch {
```
nit: fetchpatch2?

GaetanLepage replied:
Done!

samuela reviewed (approved):
Thanks for putting this together @GaetanLepage ! Any remaining blockers to merge?

GaetanLepage commented:
> Thanks for putting this together @GaetanLepage ! Any remaining blockers to merge?

Nope, if you're fine with the diff, feel free to press the button. This was a big one!
